### PR TITLE
API 6150 get duz support

### PR DESCRIPTION
**Pull request by bschofield-va**
New `/internal/user-demographics/{site}` endpoint
- Requires headers `accessCode` and `verifyCode`

User demographics ITs Includes system definition updates 
- rpcPrincipla is renamed to avCodeRpcPrincipal to better reflect the usage 
- New value "vistaSite" that is separate targets.
   - Defaults to 673 (tampa) in all environments and does not require DU updates 
   - Will use vista.site system property or VISTA_SITE environment variable if available

### CODE REVIEW

File: charon-tests/request.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "principal": {
+    "applicationProxyUser":"LHS,APPLICATION PROXY",
+    "accessCode":"123LIGHTHOUSE",
+    "verifyCode":"321LIGHTHOUSE*"
```
Comment (eclendenning-va) [inline charon-tests/request.json:5]: secret?

File: charon-tests/src/test/java/gov/va/api/lighthouse/charon/tests/UserDemographicsIT.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    var requestPath = basePath + "internal/user-demographics/{site}";
+    log.info("Running health-check for path: {}", requestPath);
+    var headers = TestClients.headers();
+    headers.remove("accessCode");
+    headers.remove("verifyCode");
+    TestClients.charon().get(headers, requestPath, SystemDefinitions.get().vistaSite()).expect(400);
+  }
+
+  @Test
+  void userDemographicsRequiresAccessVerifyCodes() {
```
Comment (eclendenning-va) [inline charon-tests/src/test/java/gov/va/api/lighthouse/charon/tests/UserDemographicsIT.java:47]: cruft?

File: charon-tests/request.json (hunk 1 above)
Comment (bschofield-va) [reply charon-tests/request.json]: I didn't meant commit that file... will delete

File: charon-tests/src/test/java/gov/va/api/lighthouse/charon/tests/UserDemographicsIT.java (hunk 2 above)
Comment (bschofield-va) [reply charon-tests/src/test/java/gov/va/api/lighthouse/charon/tests/UserDemographicsIT.java:47]: yes... will delete

**Review by eclendenning-va — APPROVED** [review]

File: charon-tests/src/test/java/gov/va/api/lighthouse/charon/tests/UserDemographicsIT.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+@Slf4j
+public class UserDemographicsIT {
+
+  @ParameterizedTest
+  @ValueSource(strings = {"/", "/charon/"})
```
Comment (jhulbert-va) [inline charon-tests/src/test/java/gov/va/api/lighthouse/charon/tests/UserDemographicsIT.java:16]: These will fail in a real environment. `/` doesn't map from the load balancer.

**Review by jhulbert-va — CHANGES_REQUESTED** [review]

File: charon-tests/src/main/java/gov/va/api/lighthouse/charon/tests/SystemDefinitions.java [hunk 4]
```diff
@@ -42,8 +42,9 @@ private static SystemDefinition lab() {
         .charon(serviceDefinition("charon", url, 443, "/charon/"))
         .clientKey(Optional.ofNullable(System.getProperty("client-key")))
         .testRpcs(rpcs())
-        .testRpcPrincipal(rpcPrincipal())
+        .avCodePrincipal(avCodeRpcPrincipal())
         .testTargets(rpcTargets())
+        .vistaSite(systemPropertyOrEnvVar("vista.site", "673"))
```
Comment (jhulbert-va) [inline charon-tests/src/main/java/gov/va/api/lighthouse/charon/tests/SystemDefinitions.java:47]: Do we want to add this configuration to the entrypoint so we can override it in different environments?

**Review by jhulbert-va — APPROVED** [review]

**Review by ilaflamme-va — APPROVED** [review]

File: charon-tests/src/main/java/gov/va/api/lighthouse/charon/tests/SystemDefinitions.java (hunk 4 above)
Comment (bschofield-va) [reply charon-tests/src/main/java/gov/va/api/lighthouse/charon/tests/SystemDefinitions.java:47]: when needed ...

**Pull request merged**